x-pack/plugins/observability/server/services/slo/reset_slo.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { resetSLOResponseSchema } from '@kbn/slo-schema';
+import {
+  getSLOTransformId,
+  SLO_DESTINATION_INDEX_PATTERN,
+  SLO_MODEL_VERSION,
+  SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+  SLO_SUMMARY_TEMP_INDEX_NAME,
+} from '../../assets/constants';
+import { SLORepository } from './slo_repository';
+import { createTempSummaryDocument } from './summary_transform/helpers/create_temp_summary';
+import { TransformManager } from './transform_manager';
+
+export class ResetSLO {
+  constructor(
+    private esClient: ElasticsearchClient,
+    private repository: SLORepository,
+    private transformManager: TransformManager
+  ) {}
+
+  public async execute(sloId: string) {
+    const slo = await this.repository.findById(sloId);
+
+    const transformId = getSLOTransformId(slo.id, slo.revision);
+    await this.transformManager.stop(transformId);
+    await this.transformManager.uninstall(transformId);
+
+    await Promise.all([this.deleteRollupData(slo.id), this.deleteSummaryData(slo.id)]);
+
+    try {
+      await this.transformManager.install(slo);
+    } catch (err) {
+      // what can we do in this case?
+      throw err;
+    }
```

Comment:
I think throwing the error is the best thing we can do at this point. As long as we bubble up the actual error message from ES so they can understand what the problem is. My guess is that if this failed (missing transform permissions OR missing transform node) the two previous calls (`stop` and `uninstall`) would have failed too. 

Maybe we don't need the `try/catch` since we're just going to throw at this point since there is nothing to rollback.